src/dawn/device_metal.mm

```diff
@@ -0,0 +1,51 @@
+// Copyright 2018 The Amber Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "device_metal.h"
+
+#include <Metal/Metal.h>
+#include <iostream>
+
+#include "amber/result.h"
+#include "dawn/dawncpp.h"
+#include "dawn_native/DawnNative.h"
+#include "dawn_native/MetalBackend.h"
+
+namespace {
```

Comment:
Any reason this isn't inside the amber::dawn namespace?

Reply:
It's not visible externally, so put it in anonymous namespace.  So it doesn't pollute some other independent definition someone might dream up some day.

Reply:
Sorry, I ment why isn't the anonymous namespace inside the amber::dawn namespace?

Reply:
It doesn't matter what the enclosing namespace is. This material is only accessible within this translation unit.  I could have declared it static instead and it would be the same.

Reply:
Changed it to inside amber::dawn, but under protest. :-)

Reply:
Thanks, I find it easier to read when you open a bunch of namespaces and then close them instead of opening and closing randomly.